### PR TITLE
CRUD API improvements

### PR DESCRIPTION
**Pull request by DifferentialOrange**
This PR introduces missing crud module options and a couple of bugfixes/doc improvements that was discovered while solving the original issue.

I didn't forget about (remove if it is not applicable):

- [x] Tests (see [documentation](https://pkg.go.dev/testing) for a testing package)
- [x] Changelog (see [documentation](https://keepachangelog.com/en/1.0.0/) for changelog format)
- [x] Documentation (see [documentation](https://go.dev/blog/godoc) for documentation style guide)

### CODE REVIEW

File: crud/result.go [hunk 1]
```diff
@@ -130,27 +132,27 @@ func (r *Result) DecodeMsgpack(d *msgpack.Decoder) error {
...
-	if msgpackIsArray(code) {
-		crudErr := newErrorMany(r.rowType)
-		if err := d.Decode(&crudErr); err != nil {
-			return err
-		}
-		retErr = *crudErr
-	} else if code != msgpcode.Nil {
-		crudErr := newError(r.rowType)
-		if err := d.Decode(&crudErr); err != nil {
+	if arrLen > 1 {
```
Comment (oleg-jukovec) [inline crud/result.go:138]: Please, update the method too:

https://github.com/tarantool/go-tarantool/blob/1f844f0c82aab458da6e28aa43bf9bf98677b25c/crud/result.go#L215

**Review by oleg-jukovec — APPROVED** [review]
Thank you for the patch!

File: crud/result.go (hunk 1 above)
Comment (DifferentialOrange) [reply crud/result.go:136]: Reworked methods to have the same structure

File: crud/result.go [hunk 2]
```diff
@@ -75,10 +75,13 @@ func (r *Result) DecodeMsgpack(d *msgpack.Decoder) error {
 		return err
 	}
 
-	if arrLen < 2 {
+	if arrLen == 0 {
 		return fmt.Errorf("array len doesn't match: %d", arrLen)
```
Comment (oleg-jukovec) [inline crud/result.go:79]: or something like this..
```suggestion
		return fmt.Errorf("unexpected empty response array")
```

Comment (DifferentialOrange) [reply crud/result.go]: Updated, added missing zero length check to bool response converter

**Review by DerekBum — APPROVED** [review]

**Pull request merged**